File: apps/android/feature/ai/domain/src/main/java/com/scribblefit/feature/ai/domain/usecase/ConfirmWorkoutUseCase.kt
package com.scribblefit.feature.ai.domain.usecase

import com.scribblefit.feature.ai.domain.model.ParsedWorkout
import com.scribblefit.feature.ai.domain.repository.WorkoutSessionRepository

/**
 * Finalizes the parsed workout session into the permanent workout ledger.
 */
class ConfirmWorkoutUseCase(
    private val sessionRepository: WorkoutSessionRepository,
    // Dependency on permanent ledger repository would go here
) {
    suspend operator fun invoke(workout: ParsedWorkout) {
        // Logic to move session to permanent storage
        sessionRepository.clearActiveSession()
    }
}

File: apps/ios/ScribbleFit/ScribbleFit/Features/AI/Domain/UseCases/ConfirmWorkoutUseCase.swift
import Foundation

/**
 * Finalizes the parsed workout session into the permanent workout ledger.
 */
public final class ConfirmWorkoutUseCase {
    private let sessionRepository: WorkoutSessionRepository
    
    public init(sessionRepository: WorkoutSessionRepository) {
        self.sessionRepository = sessionRepository
    }
    
    public func execute(workout: ParsedWorkout) async throws {
        // Logic to move session to permanent storage
        try await sessionRepository.clearActiveSession()
    }
}
